backend/app/controllers/spree/admin/quick_switch_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Spree
+  module Admin
+    class QuickSwitchController < Spree::Admin::BaseController
+      layout false
+
+      def find_object
+        quick_switch_item = Spree::Backend::Config.quick_switch_items.detect do |item|
+          item.search_triggers.include? searched_key.to_sym
+        end
+
+        if quick_switch_item
+          if object = quick_switch_item.finder.call(searched_value)
+            render(
+              json: {
+                redirect_url: quick_switch_item.url.call(object)
+              },
+              status: :ok
+            )
+          else
+            render(
+              json: {
+                message: quick_switch_item.not_found_text(searched_value)
+              },
+              status: :not_found
+            )
+          end
+        else
+          render(
+            json: {
+              message: I18n.t("invalid_query", scope: "spree.quick_switch")
+            },
+            status: :bad_request
+          )
+        end
+      end
```

Comment:
This is a really big method. Maybe move the renders to their own private methods? Like `render_redirect_to`, `render_not_found`, and `render_invalid_query`. You may have better naming than I.